clang/include/clang/Sema/Sema.h

```diff
@@ -13091,6 +13090,20 @@ void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
         return;
       }
     }
+    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelSchedulerTargetFmaxMhz ||
+        CI.getParsedKind() == ParsedAttr::AT_IntelFPGAPrivateCopies) {
+      if (ArgInt < 0) {
+        Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
+            << CI.getAttrName() << /*non-negative*/ 1;
```

Comment:
You can drop the `getAttrName()` bit -- the diagnostics engine knows how to print a `CommonAttributeInfo` object properly already.